packages/eas-cli/src/commands/release/publish.ts

```diff
@@ -0,0 +1,163 @@
+      );
+    }
+    const projectId = await ensureProjectExistsAsync({
+      accountName,
+      projectName: slug,
+    });
+
+    if (!releaseName) {
+      const validationMessage = 'release name may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ releaseName } = await promptAsync({
+        type: 'text',
+        name: 'releaseName',
+        message: 'Please enter the name of the release to publish on:',
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    if (!updateMessage) {
+      const validationMessage = 'publish message may not be empty.';
+      if (jsonFlag) {
+        throw new Error(validationMessage);
+      }
+      ({ publishMessage: updateMessage } = await promptAsync({
+        type: 'text',
+        name: 'publishMessage',
+        message: `Please enter a publication message.`,
+        initial: (await getLastCommitMessageAsync())?.trim(),
+        validate: value => (value ? true : validationMessage),
+      }));
+    }
+
+    const { id: releaseId } = await getReleaseByNameAsync({
+      appId: projectId,
+      releaseName: releaseName!,
+    });
+
+    let updateInfoGroup;
+    const assetSpinner = ora('Uploading assets...').start();
+    try {
+      const assets = collectAssets(inputDir!);
+      await uploadAssetsAsync(assets);
+      updateInfoGroup = await buildUpdateInfoGroupAsync(assets);
+      assetSpinner.succeed('Uploaded assets!');
+    } catch (e) {
+      assetSpinner.fail('Failed to upload assets');
+      throw e;
+    }
+
+    let newUpdateGroup;
+    const publishSpinner = ora('Publishing...').start();
+    try {
+      newUpdateGroup = await PublishMutation.publishUpdateGroupAsync({
+        releaseId,
+        updateInfoGroup,
+        runtimeVersion,
+        updateMessage,
+      });
+      publishSpinner.succeed('Published!');
+    } catch (e) {
+      publishSpinner.fail('Failed to published updates');
+      throw e;
+    }
+
+    if (jsonFlag) {
+      log(newUpdateGroup);
+    } else {
+      const outputMessage = new Table({
+        wordWrap: true,
+        chars: {
+          top: '',
+          'top-mid': '',
+          'top-left': '',
+          'top-right': '',
+          bottom: '',
+          'bottom-mid': '',
+          'bottom-left': '',
+          'bottom-right': '',
+          left: '',
+          'left-mid': '',
+          mid: '',
+          'mid-mid': '',
+          right: '',
+          'right-mid': '',
+          middle: ' ',
+        },
+        style: { 'padding-left': 0, 'padding-right': 0 },
+      });
+      outputMessage.push(
+        [chalk.dim('project'), `@${accountName}/${slug}`],
+        [chalk.dim('release'), releaseName],
+        [chalk.dim('runtimeVersion'), runtimeVersion],
+        [chalk.dim('updateGroupID'), newUpdateGroup.updateGroup],
+        [chalk.dim('message'), updateMessage]
+      );
+      log(outputMessage.toString());
```

Comment:
This is a combo of evan and satya's output in this discussion: https://exponent-internal.slack.com/archives/C017N0N99RA/p1607720823430200
If we are happy with how this looks, it maybe worth standardizing this throughout eas in a separate PR?

Reply:
Yeah makes sense to pull it to a separate function and use the same formatting everywhere summaries like this are shown. (In a new PR.)